### PR TITLE
Basic jax

### LINKED ISSUE #4: Issue with training loop
The code implemented in the basic_jax branch has issues in the training loop:

The training loop proceeds, but there is no improvement to the parameters.

Fix the issue, and merge into the main branch.

### PR DESCRIPTION
**Pull request by cklyne**
Implemented basic ML functionality with jax:
 - A simple dense-model class.
 - A simple estimator based around a dense neural network.
 - Log-likelihood loss.
 - Training functionality for jax networks.

Closes #4 and #8.

### CODE REVIEW

**Comment by cklyne** [comment]
closes #8

**Review by SteffenJw — APPROVED** [review]
